packages/opencensus-core/src/stats/view.ts

```diff
@@ -32,32 +33,142 @@ export class BaseView implements View {
    */
-  private rows: {[key: string]: AggregationData};
+  private rows: {[key: string]: AggregationData} = {};
+  /**
+   * A list of tag keys that represents the possible column labels
+   */
+  private columns: string[];
   /**
    * An Aggregation describes how data collected is aggregated.
    * There are four aggregation types: count, sum, lastValue and distirbution.
    */
   readonly aggregation: AggregationType;
   /** The start time for this view */
   readonly startTime: number;
+  /** The bucket boundaries in a Distribution Aggregation */
+  private bucketBoudaries?: number[];
   /**
    * The end time for this view - represents the last time a value was recorded
    */
   endTime: number;
   /** true if the view was registered */
-  registered: boolean;
+  registered = false;
 
+  /**
+   * Creates a new View instance. This constructor is used by Stats. User should
+   * prefer using Stats.createView() instead.
+   * @param name The view name
+   * @param measure The view measure
+   * @param aggregation The view aggregation type
+   * @param tagsKeys The Tags' keys that view will have
+   * @param description The view description
+   * @param bucketBoudaries The view bucket boundaries for a distribution
+   * aggregation type
+   */
   constructor(
       name: string, measure: Measure, aggregation: AggregationType,
-      tagKeys: string[], description?: string) {
-    throw new Error('Not Implemented');
+      tagsKeys: string[], description?: string, bucketBoudaries?: number[]) {
+    if (aggregation === AggregationType.DISTRIBUTION && !bucketBoudaries) {
+      throw new Error('No bucketBoundaries specified');
+    }
+    this.name = name;
+    this.description = description;
+    this.measure = measure;
+    this.columns = tagsKeys;
+    this.aggregation = aggregation;
+    this.startTime = Date.now();
+    this.bucketBoudaries = bucketBoudaries;
+  }
+
+  /**
+   * Records a measurement in the proper view's row. This method is used by
+   * Stats. User should prefer using Stats.record() instead.
+   *
+   * Measurements with measurement type INT64 will have its value truncated.
+   * @param measurement The measurement to record
+   */
+  recordMeasurement(measurement: Measurement) {
+    // Checks if measurements has all tags in views
+    for (const tagKey of this.columns) {
+      if (!Object.keys(measurement.tags).find((key) => key === tagKey)) {
+        return;
+      }
+    }
+
+    const encodedTags = this.encodeTags(measurement.tags);
+    if (!this.rows[encodedTags]) {
+      this.rows[encodedTags] = this.createAggregationData(measurement.tags);
+    }
+    Recorder.addMeasurement(this.rows[encodedTags], measurement);
+  }
+
+  /**
+   * Encodes a Tags object into a key sorted string.
+   * @param tags The tags to encode
+   */
+  private encodeTags(tags: Tags): string {
+    const encodedTags = Object.keys(tags).sort().reduce((strTags, tagKey) => {
+      return strTags + `${tagKey}:${tags[tagKey]},`;
```

Comment:
Is `:` a reasonable delineator here?

Reply:
I see. It would be safer to take advantage of the [restrictions on tag keys/values](https://github.com/census-instrumentation/opencensus-specs/blob/master/tags/TagContext.md). For example, to use non-printable characters as delineators, since both keys and values must use printable characters. That way there is guaranteed to be no name collision in weird cases where the value contains `:` or `,`.

Also, `map` + `join` might be clearer than `reduce`.